### PR TITLE
Native Fix, Removal of java.util.Locales in coreJS, and test case updates.

### PR DESCRIPTION
**Pull request by davidgbiggs**
With this update, the coreNative implementation has been updated to work without any unnecessary dependencies, java.util.Locales has been removed and the code has been more robustly tested for reliability. 

One concern is the testing environment dependence upon the user's timezone. I have tested all functions and they work in my timezone perfectly well, but some test cases must be updated to work individually with the tester's timezone.

An additional concern is edge cases like dates at the end/beginning of the year and near the epoch that remain untested. This presents some level of unreliability and is an area in which test coverage can be improved.

### CODE REVIEW

File: core/shared/src/test/scala/tests/DateFormatSpec.scala [hunk 1]
```diff
@@ -74,11 +76,11 @@ class DateFormatSpec extends WordSpec with Matchers {
     }
 
     "retrieve miscellaneous info" in {
-      gmtDate.t.timeZoneOffsetMM // can only be tested by directly redefining the method
-      gmtDate.t.timeZoneOffsetHH // can only be tested by directly redefining the method
-      gmtDate.t.timeZone // relative to the testing environment
-      gmtDate.t.z // relative to the testing environment; relies on timeZone
-      gmtDate.t.Z // relative to the testing environment; relies on timeZone
+      gmtDate.t.timeZoneOffsetMM //should be("240") enter your timezone offset minutes here
```
Comment (darkfrog26) [inline core/shared/src/test/scala/tests/DateFormatSpec.scala:79]: Why are these disabled?

Comment (davidgbiggs) [reply core/shared/src/test/scala/tests/DateFormatSpec.scala:79]: Those test cases are environment-dependent, so the user can edit them as they correspond to their respective environment. The issue is that determining the user's timezone, locale, etc. as a reference for these test cases would be redundant, because the only way we would have access to that is by using the very code we are testing.

Unless you know some third-party method of determining those factors, having the user "hard-code" their own data seems to be the best solution at this time.

Comment (darkfrog26) [reply core/shared/src/test/scala/tests/DateFormatSpec.scala:79]: We should be setting the timezone for the tests, which should be possible on all platforms.

Comment (davidgbiggs) [reply core/shared/src/test/scala/tests/DateFormatSpec.scala:79]: Okay. I'll explore this further.
